escher/scene/model.h

```diff
@@ -4,29 +4,32 @@
 
 #pragma once
 
-#include <memory>
 #include <vector>
 
+#include "escher/base/macros.h"
 #include "escher/scene/object.h"
 
 namespace escher {
 
+// The model to render.
+//
+// TODO(jeffbrown): This currently only contains a vector of objects to be
+// rendered but later on we may store additional resources used to
+// interpret the contents of the model (such as tunable parameters).
 class Model {
  public:
   Model();
-  explicit Model(std::vector<std::unique_ptr<Object>> objects);
+  explicit Model(std::vector<Object> objects);
```

Comment:
Object is copyable anyhow.  Basically the whole model is just data now.

On Wed, Jun 22, 2016, 9:01 AM Adam Barth <notifications@github.com> wrote:

> In escher/scene/model.h
> <https://github.com/domokit/waterfall/pull/18#discussion_r68081922>:
>
> >  class Model {
> >   public:
> >    Model();
> > -  explicit Model(std::vector<std::unique_ptr<Object>> objects);
> > +  explicit Model(std::vector<Object> objects);
>
> Nevermind. I forgot that std::vector supports move-only types.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/domokit/waterfall/pull/18/files/4ed633b64850c9abbcf1d718355af4c7df13ac7e#r68081922>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAGPhPff3MLBTdmIN769i3xPqWmBjWzZks5qOVxMgaJpZM4I7jJ->
> .
>